### ISSUE TITLE
waffle-shiro: problems with bodyless (zero-length) IE POST

### ISSUE THREAD

**Issue by joergmattes** [Waffle/waffle#806]
Hi
I'm using waffle-shiro, and so far it works fine for Chrome and Firefox. But on IE 11 I get a zero-length POST after a user idle time > 1 min which causes all kinds of problems. I guess this is the waffle-shiro equivalent of #167. 
I found a workaround on SO: https://stackoverflow.com/questions/40918017/ie-sends-empty-post-body-randomly-while-using-ntlm-authentication-using-angular/40918151#40918151
Just wanted to post it here in case somebody has the same problems and comes looking.

Maybe some of the main contributors would have an idea how to improve the code on SO. If yes, I'd be happy to test the idea and provide a PR.

**Comment by hazendaz** [comment]
@joergmattes The original shiro code was donated.  If you look back through our change log you should be able to discover original contributor(s) and probably could ping them on this.  Additionally, it seems what was written on SO seems to work.  If you are willing to work on a solution here for PR and can provide some form of demo to test the prior failure then what to do to fix, we can certainly get a quick patch out to solve your situation.  In the meantime, it sounds like you could use that patch on SO directly in your code and possibly get by while waiting on something official from us.

**Comment by joergmattes** [comment]
Yes, sorry if I wasn't clear: I use the patch on SO in my code and it works fine. There's no hurry.
Ok, I'll try to contact the original contributor and see what he/she thinks about that patch.

**Comment by hazendaz** [comment]
@joergmattes I think the patch can be made more internally without all the commical javadocs so it's more professional.  However, I need some help to actually fix that and the quick workaround I think is wrong.  It doesn't appear to call the original super unless I missed that and in that case, it's effectively skipping authentication for situations outside of the zero post issue.  So that solution was rather lacking.  This issue further is in a number of outstanding issues dating back a number of years so it might not be so much the shiro module to fix anyways but deeper in waffle possibly in the jna module.
